autoconfigure/collector-sqs/src/main/java/zipkin/autoconfigure/collector/sqs/ZipkinSQSCredentialsAutoConfiguration.java

```diff
@@ -0,0 +1,116 @@
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+@Configuration
+@EnableConfigurationProperties(ZipkinSQSCollectorProperties.class)
+@Conditional(ZipkinSQSCollectorAutoConfiguration.SQSSetCondition.class)
+public class ZipkinSQSCredentialsAutoConfiguration {
+
+  /** Setup {@link AWSSecurityTokenService} client an IAM role to assume is given. */
+  @Bean
+  @ConditionalOnMissingBean
+  @Conditional(STSSetCondition.class)
+  AWSSecurityTokenService securityTokenService(ZipkinSQSCollectorProperties properties) {
+    return AWSSecurityTokenServiceClientBuilder.standard()
+        .withCredentials((credentialsProvider == null) ? getDefaultCredentialsProvider(properties) : credentialsProvider)
+        .withRegion(properties.awsStsRegion)
+        .build();
+  }
+
+  @Autowired(required = false)
+  private AWSSecurityTokenService securityTokenService;
+
+  /** By default, get credentials from the {@link DefaultAWSCredentialsProviderChain */
+  @Bean
+  @ConditionalOnMissingBean
+  AWSCredentialsProvider credentialsProvider(ZipkinSQSCollectorProperties properties) {
+    if (securityTokenService != null) {
+      return new STSAssumeRoleSessionCredentialsProvider.Builder(properties.awsStsRoleArn, "zipkin-server")
+          .withStsClient(securityTokenService)
+          .build();
+    } else {
+      return getDefaultCredentialsProvider(properties);
+    }
+  }
+
+  @Autowired(required = false)
+  private AWSCredentialsProvider credentialsProvider;
+
+  private static AWSCredentialsProvider getDefaultCredentialsProvider(ZipkinSQSCollectorProperties properties) {
+    // Create credentials provider from ID and secret if given otherwise use the default provider
+    AWSCredentialsProvider provider;
+    if (!isNullOrEmpty(properties.awsAccessKeyId) && !isNullOrEmpty(properties.awsSecretAccessKey)) {
+      provider = new AWSStaticCredentialsProvider(
+          new BasicAWSCredentials(properties.awsAccessKeyId, properties.awsSecretAccessKey));
+    } else {
+      provider = new DefaultAWSCredentialsProviderChain();
```

Comment:
I think you can get the same effect by initializing the field to `new DefaultAWSCredentialsProviderChain();`

check me on that..